### PR TITLE
Bump .ruby_version to latest Ruby version (3.3.5)

### PR DESCRIPTION
**Pull request by berkos**
👋  Thanks for the review in advance. 🙏 

This PR matches locally the ruby version with the [CI ](https://github.com/rails/solid_queue/blob/main/.github/workflows/main.yml#L25).
Also corrects a couple of minor typos spotted while going through the Readme.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -252,7 +252,7 @@ class MyJob < ApplicationJob
 
   # ...
 ```
-- `key` is the only required parameter, and it can be a symbol, a string or a proc that receives the job arguments as parameters and will be used to identify the jobs that need to be limited together. If the proc returns an Active Record record, the key will be built from its class name and `id`.
+- `key` is the only required parameter, and it can be a symbol, a string or a proc that receives the job arguments as parameters and will be used to identify the jobs that need to be limited together. If the proc returns an ActiveRecord record, the key will be built from its class name and `id`.
```
Comment (rosa) [inline README.md:255]: I think [Active Record is written separate](https://guides.rubyonrails.org/active_record_basics.html) 🤔 Unless it's the specific Ruby module, but in that case I think I'd have written `ActiveRecord::Base`.

**Review by rosa — CHANGES_REQUESTED** [review]
Thanks a lot and sorry for the delay!

File: README.md (hunk 1 above)
Comment (berkos) [reply README.md]: Thanks for the review, that makes sense. I wrongly thought it was referred to the module. Corrected.

**Review by rosa — APPROVED** [review]
Thank you so much! 🙏

**Pull request merged**